### DISCUSSION TITLE
Does cvc5 have a way to understand an SMTLIB string with its Python api? (Something like Z3's .from_string())

### DISCUSSION THREAD

**Discussion by TiloRC** [cvc5/cvc5#10592] [Q&A] [answered]
In z3, you can do this:

```python
from z3 import Solver

s = Solver()

s.from_string("""
(declare-const x Int)
(declare-const y Int)

(assert (> x 2))
(assert (< y 10)) 
(assert (= (+ x y) 7))
""")

s.check()
s.model() # returns [y = 0, x = 7]
```
Is there a way to pass an SMTLIB formatted string to cvc5 in Python and get check satisfiability and get a model?

**Comment by alex-ozdemir** [comment, marked as answer]
Hi Tilo,

We haven't exposed this through the Pythonic API yet, but it is available in the base Python API. See the example on [this page](https://cvc5.github.io/docs-ci/docs-main/examples/parser.html).

You should be able to access this though the Pythonic API by accessing a (pythonic) `Solver` object's `solver` field (which is a base `Solver` object).

This is very hacky, of course. We'll work on exposing this nicely through the Pythonic API.

Cheers,
Alex
